pkg/controller/infrastructure/infraflow/flow_context.go

```diff
@@ -28,6 +29,14 @@ const (
 	defaultLongTimeout = 4 * time.Minute
 )
 
+var setSeparator = sync.OnceFunc(func() {
+	shared.Separator = "|"
```

Comment:
I'm confused, Why do you change the separator here?
Also, is setSeparator the set separator or does it set the separator?

Reply:
The `setSeparator` is supposed to change the separator used in the state once. The separator is changed from "/" to "|" because some of the strings that we save do use "/" internally and when the state is restored the slashes interfere with really restoring the state.

We can refactor the strings that we save, or escape the strings or just change the separator. Previous to this PR (and I believe after removing much of the code post-deprecation) we didn't really need to restore things from the state which is why this is only introduced now.

Reply:
> Also, is setSeparator the set separator or does it set the separator?

Is the function to set the separator once. It is executed on package import with the `init` couple lines below.